src/vmm/tests/integration_tests.rs

```diff
@@ -50,13 +47,52 @@ fn wait_vmm_child_process(vmm_pid: i32) {
 }
 
 #[test]
-fn test_dummy_vmm() {
+fn test_dummy_vmm_elf() {
     let pid = unsafe { fork() };
     let vmm_config = VMMConfig {
         kernel_config: default_kernel_config(),
         memory_config: default_memory_config(),
         vcpu_config: default_vcpu_config(),
     };
+    // Sanity check. Because the VMM runs in a separate process, if the file doesn't exist,
+    // all we see is a different exit code than 0.
+    assert!(vmm_config.kernel_config.path.as_path().exists());
+    match pid {
+        0 => {
+            set_panic_hook();
+            match VMM::try_from(vmm_config) {
+                Ok(mut vmm) => {
+                    vmm.run();
+                    // Shouldn't get here with this guest image. It will loop forever.
+                    unsafe { _exit(-1) };
+                }
+                _ => unsafe { _exit(1) },
+            }
+        }
+        vmm_pid => {
+            // Parent process: give the VMM some time, then check the exit code.
+            // It should terminate and return 0.
+            sleep(Duration::from_secs(5));
+            wait_vmm_child_process(vmm_pid);
+        }
+    }
+}
+
+#[test]
+fn test_dummy_vmm_bzimage() {
```

Comment:
Just curious, is it possible/makes any sense to have rust negative integ tests (in the future)?

Reply:
I don't see why it wouldn't make sense. It's definitely possible, we can attempt to boot a VMM with invalid parameters (and skip the valid params check, hehe).